ddtrace/tracer/log.go

```diff
@@ -107,6 +109,13 @@ func logStartup(t *tracer) {
 	if _, err := samplingRulesFromEnv(); err != nil {
 		info.SamplingRulesError = fmt.Sprintf("%s", err)
 	}
+	// Based on the apply logic in sampler.go func (*rulesSampler).applyRate,
+	// the rate limit applies only when there are rules or there is a global rate set.
+	if t.rulesSampling.globalRate > 0 || len(t.rulesSampling.rules) > 0 {
+		if t.rulesSampling != nil && t.rulesSampling.limiter != nil && t.rulesSampling.limiter.limiter != nil {
```

Comment:
Where did we go wrong that we need to have all these `nil` checks? 🙂 If this is the best possible design at the moment, maybe it warrants some helper method (e.g. `t.rulesSampling.hasRateLimiter` or `t.rulesSampling.Rate` where the latter could return a secondary `ok` boolean), then you could simply do:
```go
if r, ok := t.rulesSampling.Rate(); ok {
    info.SampleRateLimit = fmt.Sprintf("%f", r)
}
```
WDYT?